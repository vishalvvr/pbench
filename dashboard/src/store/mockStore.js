// import configureStore to create a mock store where we will dispatch our actions
import configureStore from "redux-mock-store";
// import thunk middle to make our action asynchronous
import thunk from "redux-thunk";

// initialise middlewares
const middlewares = [thunk];

// initialise MockStore which is only the configureStore method which take middlewares as its parameters
const mockStore = configureStore(middlewares);

export const mockState = {
  toastReducer: {
    alerts: [
      {
        variant: "success",
        title: "Logged in successfully!",
        key: "test1",
        message: "",
      },
      {
        variant: "success",
        title: "Saved!",
        key: "test2",
        message: "",
      },
    ],
  },
};
export const store = mockStore(mockState);
