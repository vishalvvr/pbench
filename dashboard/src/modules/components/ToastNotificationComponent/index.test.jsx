import * as reactRedux from "react-redux";

import { mockState, store } from "store/mockStore";

import React from "react";
import ToastComponent from "./index";
import { render } from "@testing-library/react";

describe("Test Toast Component", () => {
  beforeEach(() => {
    const mockDispatchFn = jest.fn();
    const useSelectorMock = jest.spyOn(reactRedux, "useSelector");
    const useDispatchMock = jest.spyOn(reactRedux, "useDispatch");

    useDispatchMock.mockReturnValueOnce(mockDispatchFn);

    useSelectorMock.mockReturnValueOnce(mockState.toastReducer);
  });
  // render the component on virtual dom
  it("Toast", () => {
    render(<ToastComponent store={store} />, {
      mockState,
    });

    expect(store.getActions()).toMatchSnapshot();
  });
});
