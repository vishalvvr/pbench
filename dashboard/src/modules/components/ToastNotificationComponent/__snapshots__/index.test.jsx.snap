// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Toast Component Toast 1`] = `Array []`;
